app/javascript/mastodon/features/compose/components/privacy_dropdown.js

```diff
@@ -272,14 +265,20 @@ class PrivacyDropdown extends React.PureComponent {
           />
         </div>
 
-        <Overlay show={open} placement={placement} target={this} container={container}>
-          <PrivacyDropdownMenu
-            items={this.options}
-            value={value}
-            onClose={this.handleClose}
-            onChange={this.handleChange}
-            placement={placement}
-          />
+        <Overlay show={open} placement={placement} flip target={this.findTarget} container={container} popperConfig={{ strategy: 'fixed' }}>
+          {({ props, placement }) => (
+            <div {...props} style={{ ...props.style, width: 350, maxWidth: '100vw' }}>
+              <div className={`dropdown-animation ${placement}`}>
+                <PrivacyDropdownMenu
+                  items={this.options}
+                  value={value}
+                  onClose={this.handleClose}
+                  onChange={this.handleChange}
+                  placement={placement}
+                />
+              </div>
+            </div>
+          )}
```

Comment:
There is another regression here in that the dropdown is displayed below the boost modal:
![image](https://user-images.githubusercontent.com/384364/210358187-0a41b414-d4c0-4e1a-8371-0d35da3ae5f9.png)

From my understanding, this is because the dropdown div is now wrapped by two more divs, so while it has a `z-index: 9999`, the wrapper doesn't.

Reply:
I resolved this by removing the z-index from `.modal-root__modal`. z-index is already set to 9999 for `.modal-root` so this does not appear to make any practical difference other than that the overlays are now working properly in the modals.